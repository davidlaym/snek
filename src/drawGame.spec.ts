import drawGame from './drawGame'

describe('draw game', () => {
  let consoleLogMock: jest.SpyInstance<void, [any?, ...any[]]>;
  let consoleClearMock: jest.SpyInstance<void, [any?, ...any[]]>;
  beforeEach(() => {
    consoleLogMock = jest.spyOn(console, 'log').mockImplementation();
    consoleClearMock = jest.spyOn(console, 'clear').mockImplementation();
  });
  afterEach(() => {
    consoleLogMock.mockRestore();
    consoleClearMock.mockRestore();
  });

  it('should be a function', () => {
    expect(typeof drawGame).toBe('function')
  });

  it('should write game state', () => {
    const initialState: GameState = {
      status: 'playing',
      snake: [[7, 1], [6, 1], [5, 1], [4, 1], [3, 1], [2, 1], [1, 1]],
      heading: 'right',
      food: [10, 10],
    };
    drawGame(initialState);
    expect(consoleLogMock.mock.calls[0]).toMatchSnapshot();
  });

  it('should write rainbow color snek', () => {
    const initialState: GameState = {
      status: 'playing',
      snake: [[11, 1], [10, 1], [9, 1], [8, 1], [7, 1], [6, 1], [5, 1], [4, 1], [3, 1], [2, 1], [1, 1]],
      heading: 'right',
      food: [10, 10],
    };
    drawGame(initialState);
    expect(consoleLogMock.mock.calls[0]).toMatchSnapshot();
  });

});