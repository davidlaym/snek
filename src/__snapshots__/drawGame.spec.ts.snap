// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draw game should write game state 1`] = `
Array [
  "┌────────────────────────────────────────────────────────────────────┐
│[32m█[0m[32m█[0m[32m█[0m[32m█[0m[32m█[0m[32m█[0m[32m█[0m                                                             │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│         [35m█[0m                                                          │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
└────────────────────────────────────────────────────────────────────┘

┌────────────────────────────┐
│  Snek Length: 07           │
└────────────────────────────┘

Move snek with arrow keys, or h,j,k,l",
]
`;

exports[`draw game should write rainbow color snek 1`] = `
Array [
  "┌────────────────────────────────────────────────────────────────────┐
│[33m█[0m[32m█[0m[33m█[0m[31m█[0m[35m█[0m[36m█[0m[33m█[0m[32m█[0m[33m█[0m[31m█[0m[35m█[0m                                                         │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│         [35m█[0m                                                          │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
│                                                                    │
└────────────────────────────────────────────────────────────────────┘

┌────────────────────────────┐
│  Snek Length: 11           │
└────────────────────────────┘

Move snek with arrow keys, or h,j,k,l",
]
`;
